flax/nnx/nnx/bridge/wrappers.py

```diff
@@ -55,43 +56,52 @@ def _functional_constructor(*args: tp.Any, **kwargs: tp.Any) -> Functional[M]:
   return _functional_constructor
 
 
-class LinenWrapper(Module):
+def shaped_init(module: Module, *args, **kwargs):
+  """To trigger init of all `LinenToNNX` module variables and return a wholesome state."""
+  assert callable(module)
+  _ = module(*args, **kwargs)
+  return nnx.split(module)
+
+
+class LinenToNNX(Module):
   def __init__(
     self,
     module: linen.Module,
-    *args: tp.Any,
     rngs: tp.Optional[Rngs] = None,
-    **kwargs: tp.Any,
   ):
     self.module = module
-
-    _rngs = (
-      {name: stream.key.raw_value for name, stream in rngs.items()}
-      if rngs
-      else {}
-    )
-    # rename default to params
-    if 'params' not in _rngs and 'default' in _rngs:
-      _rngs['params'] = _rngs['default']
-      del _rngs['default']
-
-    variables = module.init(_rngs, *args, **kwargs)
-
-    self.states = {
-      collection: variableslib.variable_type(collection)(value)
-      for collection, value in variables.items()
-    }
+    self.rngs = rngs
 
   def __call__(
     self, *args: Any, rngs: tp.Optional[Rngs] = None, **kwargs: Any
   ) -> Any:
-    _rngs = (
-      {name: stream.key.value for name, stream in rngs.items()} if rngs else {}
-    )
 
-    variables = {
-      collection: value.value for collection, value in self.states.items()
-    }
+    # Shape-based lazy init of the flax variables
+    if not rngs:
+      rngs = self.rngs
+    if not hasattr(self, 'states'):
```

Comment:
Use `self._object__state.initializing` instead, see above.
```suggestion
    if self._object__state.initializing:
```

Reply:
Done.